libs/tools/barista/src/transform.ts

```diff
@@ -236,6 +237,66 @@ export const relativeUrlTransformer: BaPageTransformer = async (source) => {
   return transformed;
 };
 
+/** Add contentSection to headlines to be able to diferentiate between headlines */
+export const sectionTransformer: BaPageTransformer = async (source) => {
+  const transformed = { ...source };
+
+  transformed.content = runWithCheerio(source.content, ($) => {
+    $('h2, h3').each((_, element) => {
+      let sectionId = $(element).attr('id');
+      if (!sectionId) {
+        sectionId = $(element).text().replace(' ', '-');
+      }
+      // console.log(sectionId);
+      $(element.attribs).append(
+        $(element).attr('contentSection', ''),
+        $(element).attr('id', sectionId),
+      );
+    });
+  });
+
+  return transformed;
+};
+
+/** generates a toc json containing headlines with children */
+export const tableOfContentGenerator: BaPageTransformer = async (source) => {
+  const transformed = { ...source };
+  let toc: TableOfContents[] = [];
+  if (source.toc) {
+    // generate TOC and at to source
+    // Find Headlines and corresponding subheadlines (h2 > h3)
+    transformed.content = runWithCheerio(source.content, ($) => {
+      const headlines = $('h2, h3');
```

Comment:
```suggestion
      const headlines = $('.ba-page h2, .ba-page h3');
```
If we scope this to the body of the page - we avoid side-effects of h2 and h3 appearing in the toc that are maybe in the footer or header section outside the ba-page. And we can remove the section transformer.

Reply:
There's no need for that. The `source.content` coming in only includes the headlines from the corresponding components markdown.
Sorry, I should have clarified that.